keras_cv/layers/preprocessing/grayscale.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from tensorflow.keras import layers, backend
+
+
+class Grayscale(layers.Layer):
+    """Grayscale class for transforming RGB image to Grayscale image. The expected images 
+    should be [0-255] pixel ranges.
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Args:
+        num_output_channels. It represent the output channel number after transformation. Valid values are 1 and 3.
+        (..., height, width, 1) for num_output_channels = 1 .
```

Comment:
explain these comments here.  I see what you're going for but be sure to clearly explain this.  Also indent one more time.